core/core/include/test/common/harness.h

```diff
@@ -0,0 +1,60 @@
+/* The MIT License (MIT)
+*
+* Copyright (c) 2017 Andrew Yeung/Bhavya Kashyap <ab.deveng@gmail.com>
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE. */
+
+#pragma once
+
+#include <stdint.h>
+
+#include <service/logger.h>
+#include <test/common/evaluator.h>
+
+class ITime;
+
+class CTestHarness
+{
+public:
+    CTestHarness(CEvaluator&, ILogger&, ITime&);
+    ~CTestHarness();
+
+protected:
+    // Helpers: Evaluator
+    CEvaluator::RESULT Validate(CEvaluator::EVALUATION<int8_t>);
+    CEvaluator::RESULT Validate(CEvaluator::EVALUATION<int16_t>);
+    CEvaluator::RESULT Validate(CEvaluator::EVALUATION<int32_t>);
```

Comment:
These are helpers that pass through directly to the CEvaluator member variable. Instead of having the harness implement CEvaluator which would expose the harness to be abused as a generic evaluator, it takes an evaluator object and provides derived classes internal methods to access that evaluator (through these protected methods). That way, only the intended client of this functionality gets access to it (ie. the derived class).